modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/view.jsp

```diff
@@ -139,22 +139,15 @@ AssetListManagementToolbarDisplayContext assetListManagementToolbarDisplayContex
 		<c:otherwise>
 			<liferay-frontend:empty-result-message
 				actionDropdownItems="<%= assetListDisplayContext.isShowAddAssetListEntryAction() ? assetListDisplayContext.getAddAssetListEntryDropdownItems() : null %>"
-				componentId="emptyResultMessageComponent"
-				defaultEventHandler="emptyResultMessageComponentDefaultEventHandler"
 				description="<%= assetListDisplayContext.getEmptyResultMessageDescription() %>"
 				elementType='<%= LanguageUtil.get(request, "collections") %>'
+				propsTransformer="js/EmptyResultMessagePropsTransformer"
+				propsTransformerServletContext="<%= application %>"
```

Comment:
Do we need  `propsTransformerServletContext`? So far I have not encountered a case where we would need it. What is the case where we do?

Reply:
Yes we do.
Otherwise it will require the JS from the tag's module instead of the current module

Reply:
Why don't we need it with `clay:*` tags? Are we doing some magic in `frontend-taglib-clay`?

Reply:
Yes, we are doing something [here](https://github.com/liferay/liferay-portal/blob/67e57d147b6b7f13bb101be79d43bd0a95148ab1/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/internal/servlet/taglib/BaseContainerTag.java#L257-L261) and [here](https://github.com/liferay/liferay-portal/blob/67e57d147b6b7f13bb101be79d43bd0a95148ab1/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/internal/servlet/taglib/BaseContainerTag.java#L319-L338): 

My understanding is that with `BaseContainerTag` (and most `<clay:*>` tags): `propsTransformerServletContext` always has a value, and if a `propsTransformer` attribute is passed we always try to load it from that "context". 

Let me know if that makes sense.

Reply:
Yeah, makes sense. Thanks for the explanation!

Reply:
No worries.

Reply:
So, you can think of `servletContext` as the JSP that is currently being rendered. The value more or less has a content by default, so when we try to locate a `.js` file from a `.jsp` we look into that `servletContext`.

There are times like this where the indirection level makes `servletContext` not match what you want. 

For example in here you want to load the `propsTransformer.js` file in the context of `asset-list-web` module, but when the code runs it will do so in the context of the `frontend-taglib` module (because it is passed through the `liferay-frontend:empty-result-message`. That means it will try to load something like `/o/frontend-taglib/js/EmptyResultPropsTransformer.js` instead of `/o/asset-list-web/js/EmptyResultPropsTransformer.js`.

So, in some cases where we're passing strings that need to be resolved in context we need to be able to specify in which context specifically this will happen.